src/util/RandomEvictionCache.h

```diff
@@ -145,56 +150,51 @@ template <typename K, typename V> class RandomEvictionCache
         return !miss;
     }
 
+    // `clear` does not throw
     void
     clear()
     {
         mValuePtrs.clear();
         mValueMap.clear();
     }
 
-    template <typename F>
+    // `erase_if` offers basic exception safety guarantee. If it throws an
+    // exception, then the cache may or may not be modified.
     void
-    erase_if(F const& f)
+    erase_if(std::function<bool(V const&)> const& f)
     {
         for (size_t i = 0; i < mValuePtrs.size(); ++i)
         {
-            map_value_type*& vp = mValuePtrs.at(i);
-            while (mValuePtrs.size() != 0 && f(vp->second.mValue))
+            MapValueType*& vp = mValuePtrs.at(i);
+            while (mValuePtrs.size() != i && f(vp->second.mValue))
             {
+                // - `erase(k)` does not throw an exception unless that
+                // exception is thrown by the container’s Hash or Pred object
+                // (if any)
+                // - `std::swap` does not throw
+                // - `std::unordered_map::pop_back` does not throw
                 mValueMap.erase(vp->first);
                 std::swap(vp, mValuePtrs.back());
                 mValuePtrs.pop_back();
             }
         }
     }
 
+    // `get` offers no exception safety guarantee. If it throws an exception,
```

Comment:
I think `get` has the strong exception safety guarantee. What led you to believe otherwise?

Reply:
Originally, I was concerned with the assignment of `mLastAccess`. Looking closer, although the standard does not provide any guarantees on that front, since the type is `uint64_t`, and there's no type conversion, I _think_ it actually does not throw. This is now updated (and squashed)